### PR TITLE
[webview_flutter_android][webview_flutter_wkwebview] Adds platform implementations for onHttpError

### PR DESCRIPTION
**Pull request by bparrishMines**
Copy of https://github.com/flutter/packages/pull/3695 since it doesn't contain permission to edit from contributors.

Part of https://github.com/flutter/flutter/issues/39502

Full PR https://github.com/flutter/packages/pull/3278

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/webview_flutter/webview_flutter_wkwebview/lib/src/foundation/foundation.dart [hunk 1]
```diff
@@ -221,6 +221,20 @@ class NSErrorUserInfoKey {
       'NSErrorFailingURLStringKey';
 }
 
+/// A URL load request that is independent of protocol or URL scheme.
```
[the remark this answers is not in the document]
Comment (bparrishMines) [reply packages/webview_flutter/webview_flutter_wkwebview/lib/src/foundation/foundation.dart]: Ah good catch. It looks like this class was accidentally using the documentation from [NSURLRequest](https://developer.apple.com/documentation/foundation/nsurlrequest?language=objc). I updated it with the right class.

File: packages/webview_flutter/webview_flutter_wkwebview/pigeons/web_kit.dart [hunk 2]
```diff
@@ -132,6 +132,20 @@ class WKNavigationActionPolicyEnumData {
...
 
+/// Mirror of WKNavigationResponsePolicy.
+///
+/// See https://developer.apple.com/documentation/webkit/wknavigationactionpolicy?language=objc.
+enum WKNavigationResponsePolicyEnum {
+  allow,
+  cancel,
+}
+
+// TODO(bparrishMines): Enums need be wrapped in a data class because thay can't
```
[the remark this answers is not in the document]
Comment (bparrishMines) [reply packages/webview_flutter/webview_flutter_wkwebview/pigeons/web_kit.dart]: updated. The original PR was probably created before the issue was fixed.

**Comment by jmagman** [comment]
@bparrishMines is this ready to merge, or are you waiting for more reviewer feedback?

**Comment by bparrishMines** [comment]
It's ready, I just forgot about adding the label. I also wanted to respond to @stuartmorgan about including a `is<FunctionName>Overriden` feature in the pigeon wrapper once the initial implementation is done.

**Pull request merged**